File: src/main/java/no/jstien/bikeapi/tsdb/read/TSDBException.java
package no.jstien.bikeapi.tsdb.read;

public class TSDBException extends RuntimeException {
    private int statusCode;

    public TSDBException(String message) {
        super(message);
        statusCode = -1;
    }

    public TSDBException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public int getStatusCode() {
        return statusCode;
    }
}

File: src/main/java/no/jstien/bikeapi/tsdb/read/TSDBException.kt
package no.jstien.bikeapi.tsdb.read

class TSDBException : RuntimeException {
    var statusCode: Int = 0
        private set

    constructor(message: String) : super(message) {
        statusCode = -1
    }

    constructor(message: String, statusCode: Int) : super(message) {
        this.statusCode = statusCode
    }
}
